src/EPR.Payment.Mopup.Common.Data/Repositories/PaymentsRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using EPR.Payment.Mopup.Common.Constants;
+using EPR.Payment.Mopup.Common.Data.Interfaces;
+using EPR.Payment.Mopup.Common.Data.Interfaces.Repositories;
+using EPR.Payment.Mopup.Common.Enums;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+
+namespace EPR.Payment.Mopup.Common.Data.Repositories
+{
+    public class PaymentsRepository : IPaymentsRepository
+    {
+        private readonly IAppDbContext _dataContext;
+        private readonly IConfiguration _configuration;
+        public PaymentsRepository(
+            IAppDbContext dataContext,
+            IConfiguration configuration)
+        {
+            _dataContext = dataContext;
+            _configuration = configuration;
+        }
+
+        public async Task UpdatePaymentStatusAsync(DataModels.Payment? entity, CancellationToken cancellationToken)
+        {
+            if (entity == null)
+            {
+                throw new ArgumentException(ExceptionMessages.InvalidInputToUpdatePaymentError);
+            }
+
+            entity.UpdatedDate = DateTime.Now;
```

Comment:
Can this datetime be unit testable ?

Reply:
@zozanKirkutay please test the value of UpdatedDate. Something like below in UpdatePaymentStatusAsync_ValidInput_ShouldComplete

            ```
using (new AssertionScope())
            {
                **request.UpdatedDate.Should().BeCloseTo(DateTime.Now, TimeSpan.FromSeconds(3));** 
                _dataContextMock.Verify(c => c.Payment.Update(It.Is<Common.Data.DataModels.Payment>(s => s.UserId == userId && s.OrganisationId == organisationId)), Times.Once());
                _dataContextMock.Verify(c => c.SaveChangesAsync(It.IsAny<CancellationToken>()), Times.Exactly(1));
            }
```

Reply:
I already added test for this part as the following
 request.UpdatedDate.Should().BeCloseTo(DateTime.Now, TimeSpan.FromSeconds(1));